src/lib/ndpi_main.c

```diff
@@ -10573,3 +10573,14 @@ void *ndpi_get_user_data(struct ndpi_detection_module_struct *ndpi_str)
     return ndpi_str->user_data;
   return NULL;
 }
+
+/* ******************************************************************** */
+
+int tpkt_verify_hdr(const struct ndpi_packet_struct * const packet)
```

Comment:
Should this function better reside in another source file like `ndpi_utils.c` ? I am unsure.

Reply:
I dunno. I've seen all the packet stuff is mostly in ndpi_main.c

Reply:
No strong opinion, but `ndpi_utils.c` is probably a better place